application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,417 @@
+        MessageEmbed embed = constructEmbedMessage(rssItem, feedConfig).build();
+        textChannels.forEach(channel -> channel.sendMessageEmbeds(List.of(embed)).queue());
+    }
+
+    /**
+     * Updates the last posted date to the database for the specified RSS feed configuration.
+     * <p>
+     * This will insert a <b>new</b> entry to the database if the provided {@link RssFeedRecord} is
+     * null.
+     *
+     * @param feedConfig the RSS feed configuration
+     * @param rssFeedRecord the record representing the RSS feed, can be null if not found in the
+     *        database
+     * @param lastPostedDate the last posted date to be updated
+     *
+     * @throws DateTimeParseException if the date cannot be parsed
+     */
+    private void updateLastDateToDatabase(RSSFeed feedConfig, @Nullable RssFeedRecord rssFeedRecord,
+            ZonedDateTime lastPostedDate) {
+        DateTimeFormatter dateTimeFormatter =
+                DateTimeFormatter.ofPattern(feedConfig.dateFormatterPattern());
+        String lastDateStr = lastPostedDate.format(dateTimeFormatter);
+
+        if (rssFeedRecord == null) {
+            database.write(context -> context.newRecord(RSS_FEED)
+                .setUrl(feedConfig.url())
+                .setLastDate(lastDateStr)
+                .insert());
+            return;
+        }
+
+        database.write(context -> context.update(RSS_FEED)
+            .set(RSS_FEED.LAST_DATE, lastDateStr)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .executeAsync());
+    }
+
+    /**
+     * Attempts to get a {@link ZonedDateTime} from an {@link Item} with a provided string date time
+     * format.
+     * <p>
+     * If either of the function inputs are null or a {@link DateTimeParseException} is caught, the
+     * oldest-possible {@link ZonedDateTime} will get returned instead.
+     *
+     * @param item The {@link Item} from which to extract the date.
+     * @param dateTimeFormat The format of the date time string.
+     * @return The computed {@link ZonedDateTime}
+     * @throws DateTimeParseException if the date cannot be parsed
+     */
+    private static ZonedDateTime getDateTimeFromItem(Item item, String dateTimeFormat)
+            throws DateTimeParseException {
+        String pubDate = item.getPubDate().orElse(null);
+
+        if (pubDate == null || dateTimeFormat == null) {
+            return ZONED_TIME_MIN;
+        }
+
+        return getZonedDateTime(pubDate, dateTimeFormat);
+    }
+
+    /**
+     * Checks if the dates between an RSS item and the provided config match.
+     *
+     * @param rssItem the RSS feed item
+     * @param feedConfig the RSS feed configuration containing the date formatter pattern
+     * @return true if the date format is valid, false otherwise
+     */
+    private static boolean isValidDateFormat(Item rssItem, RSSFeed feedConfig) {
+        String firstRssFeedPubDate = rssItem.getPubDate().orElse(null);
```

Comment:
Same as above, you shouldn't use `.orElse(null)` and then checking if it is null. Instead, store the Optional and check if it is empty.

Reply:
Done!